cmd/client_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2020 Google LLC // // Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove (@evanSpendlove)
+//
+// Client_test tests the client.go file to ensure that the gRPC client for connecting
+// to Cloudprober works as expected. This client is used for adding and removing probes.
+
+package cmd
+
+import (
+	"github.com/golang/glog"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/cloudprober"
+	"github.com/google/cloudprober/examples/extensions/myprober/myprobe"
+	probes_configpb "github.com/google/cloudprober/probes/proto"
+	targetspb "github.com/google/cloudprober/targets/proto"
+	"github.com/googleinterns/step224-2020/hermes"
+)
+
+const (
+	cpCfg     string = "grpc_port:9314" // Cloudprober config
+	rpcServer string = "localhost:9314" // Cloudprober gRPC server address
+)
+
+// TODO: Make these proper table-driven tests.
+var (
+	testProbes = []string{
+		"testExtension0",
+		"testExtension1"}
+)
+
+func setupCloudproberAndClient() (*hermes.Hermes, *CloudproberClient) {
+	testHermes := &hermes.Hermes{}
+	err := testHermes.InitialiseCloudproberFromConfig(cpCfg)
+	if err != nil {
+		glog.Fatalf("Cloudprober could not be initialised, err: %v", err)
+	}
+
+	cloudprober.Start(testHermes.Ctx)
+
+	client, errClient := NewClient(rpcServer)
+	if errClient != nil {
+		glog.Fatalf("Cloudprober gRPC Client could not be initialised: %v", errClient)
+	}
+
+	return testHermes, client
+}
+
+func teardownCloudproberAndClient(testHermes *hermes.Hermes, client *CloudproberClient) {
+	// TODO: Find a safe way to close down Cloudprober and restart it.
```

Comment:
All to-do's must have an owner. Which can be either an issue or a human (a.k.a. their github username)
```suggestion
	// TODO(evanSpendlove): Find a safe way to close down Cloudprober and restart it.
```

Reply:
I have added an owner to all TODOs in this PR.